aten/src/ATen/native/TensorShape.cpp

```diff
@@ -356,10 +356,24 @@ static Tensor cat_sparse(TensorList tensors, int64_t dim) {
 }
 
 Tensor cat(TensorList tensors, int64_t dim) {
+  std::vector<Tensor> temp_promote;
+  if (tensors.size() > 1) { 
+    ScalarType high_type = tensors[0].scalar_type();
+    for (size_t i = 1; i < tensors.size(); ++i) {
+      high_type = promoteTypes(high_type, tensors[i].scalar_type());
```

Comment:
nit: I think it would be preferable to use resultType which accepts tensors instead of scalar types. The result is the same because cat doesn't accept 0-dim tensors anyway, but using the tensor function enforces consistency in the implementation.

Reply:
I could only find `result_type` taking the list of tensors to get the promoted dtype.